### PR TITLE
For upstream 3

### PR DESCRIPTION
**Pull request by LiuZhiguang001**
Three patch set.
1. parse PCI host bridge hob
2. parse acpi hob.
3. parse smbios hob.

### CODE REVIEW

File: MdeModulePkg/Universal/Acpi/AcpiTableDxe/AcpiTableProtocol.c [hunk 1]
```diff
@@ -552,13 +557,17 @@ AddTableToList (
...
-    Status = gBS->AllocatePages (
-                    AllocateMaxAddress,
-                    EfiACPIMemoryNVS,
-                    EFI_SIZE_TO_PAGES (CurrentTableList->TableSize),
-                    &AllocPhysAddress
-                    );
+    if (IsFromHob){
+      AllocPhysAddress = (UINTN)Table;
+    } else {
+      ASSERT ((EFI_PAGE_SIZE % 64) == 0);
```
Comment (mauricema) [inline MdeModulePkg/Universal/Acpi/AcpiTableDxe/AcpiTableProtocol.c]: Is this required  ?   EFI_PAGE_SIZE is already page aligned.

**Review by mauricema — APPROVED** [review]
Looks good to me.

File: MdeModulePkg/Universal/Acpi/AcpiTableDxe/AcpiTableProtocol.c (hunk 1 above)
Comment (LiuZhiguang001) [reply MdeModulePkg/Universal/Acpi/AcpiTableDxe/AcpiTableProtocol.c]: Thanks for catching this.
I agree it is not needed, and will remove it

**Review by gdong1 — APPROVED** [review]

File: MdeModulePkg/Include/Guid/SmBiosTableHob.h [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+#ifndef _SMBIOS_TABLE_HOB_H_
+#define _SMBIOS_TABLE_HOB_H_
+
+#include <Uefi.h>
+
+#pragma pack(1)
+
+typedef struct {
+  EFI_PHYSICAL_ADDRESS          SmBiosTableAddress;
```
Comment (niruiyu) [inline MdeModulePkg/Include/Guid/SmBiosTableHob.h]: How about "SmBiosEntryPoint"?
For ACPI, we use "Rsdp".

**Review by niruiyu — APPROVED** [review]

**Pull request merged**